src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpInstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,5 +20,77 @@ namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
     public sealed class CSharpInstantiateArgumentExceptionsCorrectlyFixer : InstantiateArgumentExceptionsCorrectlyFixer
     {
+        protected override void PopulateCodeFix(CodeFixContext context, Diagnostic diagnostic, string paramPositionString, SyntaxNode node)
+        {
+            if (node is ObjectCreationExpressionSyntax creation)
+            {
+                int paramPosition = int.Parse(paramPositionString, CultureInfo.InvariantCulture);
+                CodeAction? codeAction = null;
+                if (creation.ArgumentList.Arguments.Count == 1)
+                { // Add null message
+                    codeAction = CodeAction.Create(
+                        title: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyChangeToTwoArgument,
+                        createChangedDocument: c => AddNullMessageToArgumentList(context.Document, creation, c),
+                        equivalenceKey: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyChangeToTwoArgument);
+                }
+                else
+                { // Swap message and paramete name
+                    codeAction = CodeAction.Create(
+                        title: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyFlipArgumentOrder,
+                        createChangedDocument: c => SwapArgumentsOrder(context.Document, creation, paramPosition, creation.ArgumentList.Arguments.Count, c),
+                        equivalenceKey: MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyFlipArgumentOrder);
+                }
+                context.RegisterCodeFix(codeAction, diagnostic);
+            }
+        }
+
+        private static async Task<Document> SwapArgumentsOrder(Document document, ObjectCreationExpressionSyntax creation, int paramPosition, int argumentCount, CancellationToken token)
```

Comment:
Would this work if you had an `IObjectCreationOperation creation` here?

Reply:
not sure, will try

Reply:
it worked with little alterations